event.go

```diff
@@ -176,8 +208,33 @@ func (eb *EventBuilder) Build(eventID string, now time.Time, origin ServerName,
 		return
 	}
 
-	result.eventJSON = eventJSON
-	if err = json.Unmarshal(eventJSON, &result.fields); err != nil {
+	result.roomVersion = roomVersion
+
+	switch eventFormat {
+	case EventFormatV1:
+		result.eventJSON = eventJSON
+		var fields eventFormatV1Fields
+		if err = json.Unmarshal(eventJSON, &fields); err != nil {
+			return
+		}
+		fields.fixNilSlices()
+		result.fields = fields
+	case EventFormatV2:
+		if result.eventJSON, err = sjson.DeleteBytes(eventJSON, "event_id"); err != nil {
+			return
+		}
+		var fields eventFormatV2Fields
+		if err = json.Unmarshal(eventJSON, &fields); err != nil {
```

Comment:
Why do we unmarshal the eventJSON which contains `event_id`? Shouldn't we be using `result.eventJSON`?

Reply:
Fixed.

Reply:
Fixed.